src/net/download.rs

```diff
@@ -316,7 +355,10 @@ impl BlockDownloader {
             if url_str.len() == 0 {
                 continue;
             }
-            let url = url_str.parse_to_block_url()?; // NOTE: should always succeed, since a UrlString shouldn't decode unless it's a valid URL or the empty string
+            let url = match url_str.parse_to_block_url() {
```

Comment:
This might not work for attachments, because `parse_to_block_url` is very particular about blocks -- it will not allow a URL to have a query string or fragment, for example.

Reply:
Change reverted https://github.com/blockstack/stacks-blockchain/pull/2013/commits/b9f70bff21a12f2b584c57ca277b761db85df497

Reply:
Just to be clear -- a URL to an attachment will never have a query string or fragment, right?

Reply:
Correct!